nvflare/private/fed/app/client/client_train.py

```diff
@@ -109,24 +109,26 @@ def main():
         federated_client.use_gpu = False
         federated_client.config_folder = config_folder
 
+        client_engine = ClientEngine(federated_client, args, rank)
+
         while federated_client.cell is None:
             print("Waiting client cell to be created ....")
             time.sleep(1.0)
 
-        federated_client.register()
+        with client_engine.new_context() as fl_ctx:
+            client_engine.fire_event(EventType.SYSTEM_BOOTSTRAP, fl_ctx)
+
+            client_engine.fire_event(EventType.BEFORE_CLIENT_REGISTER, fl_ctx)
+            register_data = fl_ctx.get_prop(FLContextKey.CLIENT_REGISTER_DATA, {})
```

Comment:
So, the custom code from users should set "FLContextKey.CLIENT_REGISTER_DATA" when they handling event "EventType.SYSTEM_BOOTSTRAP" or "EventType.BEFORE_CLIENT_REGISTER"?

Reply:
The application will use the event handlers to listen the SYSTEM_BOOTSTRAP and BEFORE_CLIENT_REGISTER events, place their own authentication related data in the CLIENT_REGISTER_DATA, which is a dict. Server side will extract and access the same CLIENT_REGISTER_DATA, then plugin the event handlers to perform their authentication operations as their need.

Reply:
thanks, that's clear enough, we need to convey this to our users